### DISCUSSION TITLE
How can user change secret key in new console

### DISCUSSION THREAD

**Discussion by markovendelin** [minio/minio#12649] [Q&A] [answered]
With the replacement of WEB interface shown to the users with console, it is not clear how users can change their secret key (aka password). The corresponding feature seems to be missing when logging in as a regular (non-admin) user.

Version: 
RELEASE.2021-07-08T01-15-01Z

**Comment by harshavardhana** [comment]
Do you not see the https://play.min.io:9443/account tab? and `Change Password` button?

**Reply by harshavardhana** [reply]
// cc @Alevsk

**Reply by markovendelin** [reply]
Thanks for moving it into discussion, at leas for now. When I login as a regular user, there are no accounts listed under "account" nor I can change the password.

**Reply by markovendelin** [reply]
As there was a suggestion to check what it is on https://play.min.io:9443/account , I did it and, indeed, there is a corresponding button on accounts page. For some reason, I am missing it in our setup (see screenshot). Any tips on how to debug it? 
![no_changepwd_button](https://user-images.githubusercontent.com/3975391/124986220-0dde4300-e044-11eb-88e9-8a9afc8f6642.png)

**Reply by harshavardhana** [reply]
I will check

**Reply by markovendelin** [reply]
Thank you very much!

**Reply by harshavardhana** [reply]
![screenshot](https://user-images.githubusercontent.com/622699/125035947-46335080-e047-11eb-8ea8-4154a7ab2503.png)

Seems to work fine

**Reply by markovendelin** [reply]
I can see "Change Password" for a user with the admin privileges. However, it is missing for a user with policy

```JSON
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "s3:*"
            ],
            "Resource": [
                "arn:aws:s3:::${aws:username}-*"
            ]
        }
    ]
}
```

This is a setup using MinIO docker image. Don't know if it makes any difference, the user was created as a "User" not a service account. This is for the both users - admin and regular ones. Don't know what service accounts are for, unfortunately.

For reference, admin user has policy

```JSON
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "admin:*"
            ]
        },
        {
            "Effect": "Allow",
            "Action": [
                "s3:*"
            ],
            "Resource": [
                "arn:aws:s3:::*"
            ]
        }
    ]
}
```
For admin account all works.

**Reply by harshavardhana** [reply]
> I can see "Change Password" for a user with the admin privileges. However, it is missing for a user with policy

That is missing because your policy doesn't allow any `admin:*`  actions @markovendelin

**Reply by markovendelin** [reply]
In this respect, it is a regression when compared to old Web UI. The old version allowed users to change their secret key. I also don't see any specific admin policy that would allow me to let users to change their passwords (https://docs.min.io/minio/baremetal/security/IAM/iam-policies.html#mc-admin-policy-action-keys)

**Reply by harshavardhana** [reply]
> In this respect, it is a regression when compared to old Web UI. The old version allowed users to change their secret key. I also don't see any specific admin policy that would allow me to let users to change their passwords (https://docs.min.io/minio/baremetal/security/IAM/iam-policies.html#mc-admin-policy-action-keys)

You need to add `admin:CreateUser`  but that is just policy interpretation on the client side, in-fact for a user changing their own password is implicit. So it needs to be fixed in https://github.com/minio/console

**Reply by markovendelin** [reply]
Thank you very much! As I cannot limit usernames created by users themselfs (https://github.com/minio/minio/issues/12401), I don't want users to be able to create new ones. Should I file an issue in console repo?

**Reply by markovendelin** [reply]
I opened an issue in Console repository: https://github.com/minio/console/issues/857

**Comment by markovendelin** [comment, marked as answer]
For reference: fixed in https://github.com/minio/console/pull/861 by @harshavardhana .
